utils/onnxmlirrun.py

```diff
@@ -0,0 +1,174 @@
+﻿#!/usr/bin/python
+# Copyright 2019-2023 The IBM Research Authors.
+
+import os
+import sys
+import onnx
+import time
+import signal
+import subprocess
+import numpy as np
+import tempfile
+
+from onnx import numpy_helper
+from onnx.mapping import TENSOR_TYPE_TO_NP_TYPE
+from collections import OrderedDict
+
+# This file provide utility to compile and run onnx model with onnx-mlir,
+# in an existing python env, such as Pytorch, tensorflow or SciKit learn.
+# The interface is delibrately designed similar to onnxruntime to reduce user's
+# burden of learning and code change.
+# Lots of code is inherited from utils/RunONNXModel.py, which is a 
+# "standalone" python script to use onnx-mlir compiler.
+# In future, this file will evolved to be part of onnx-mlir python package. 
+# An example:
+
+"""
+import onnxmlirrun
+import numpy as np
+
+a = np.random.rand(3, 4, 5).astype(np.float32)
+b = np.random.rand(3, 4, 5).astype(np.float32)
+session = onnxmlirrun.InferenceSession("test_add.onnx")
+outputs = session.run(None, {"a": a, "b":b})
+print(outputs)
+"""
+
+
+
+class InferenceSession:
+
+    def __init__(self, model_path):
+
+        # Initialize status
+
+        self.compiled = False
+        self.loaded = False
+
+        # Initialize parameters
+
+        self.VERBOSE = os.environ.get('VERBOSE', False)
+        self.input_model_path = model_path
+
+        # name for the compiled library in temporary directory
+
+        self.temp_lib_name = 'model'
+        if not os.environ.get('ONNX_MLIR_HOME', None):
+            raise RuntimeError('Environment variable ONNX_MLIR_HOME is not set, please set it to the path to the HOME directory for onnx-mlir. The HOME directory for onnx-mlir refers to the parent folder containing the bin, lib, etc sub-folders in which ONNX-MLIR executables and libraries can be found, typically `onnx-mlir/build/Debug`'
+                               )
+
+        self.ONNX_MLIR_EXENAME = 'onnx-mlir'
+        if sys.platform == 'win32':
+            self.ONNX_MLIR_EXENAME = 'onnx-mlir.exe'
+
+        # Compiler package related parameters.
+        # Should be changed when package is installed
+
+        self.ONNX_MLIR = os.path.join(os.environ['ONNX_MLIR_HOME'],
+                'bin', self.ONNX_MLIR_EXENAME)
+        self.RUNTIME_DIR = os.path.join(os.environ['ONNX_MLIR_HOME'],
+                'lib')
+        sys.path.append(self.RUNTIME_DIR)
+        try:
+            from PyRuntime import OMExecutionSession
+        except ImportError:
+            raise ImportError('Looks like you did not build the PyRuntime target, build it by running `make PyRuntime`.You may need to set ONNX_MLIR_HOME to `onnx-mlir/build/Debug` since `make PyRuntime` outputs to `build/Debug` by default'
+                              )
+
+    def compile(self):
```

Comment:
It's good if users can pass compiler options into this function.

Reply:
`compile()` will not be directly used by user. Compiler options can be passed when the session is initialized.